extensions/typescript-language-features/src/languageFeatures/diagnostics.ts

```diff
@@ -148,6 +151,87 @@ class DiagnosticSettings {
 	}
 }
 
+class DiagnosticsTelemetryManager extends Disposable {
+
+	private readonly _diagnosticCodesMap = new Map<number, number>();
+	private readonly _diagnosticTimeoutsMap = new Map<URI, NodeJS.Timeout | undefined>();
```

Comment:
I suggest we have a single timeout that works across all documents. The idea would be to update once the user stops for a few seconds.

Reply:
Yep removed the map, there is not only one timeout. I think the little issue with that is that if the user types something and within 10 seconds switches to a new document and types inside of that second one, the diagnostics will not be updated in the first. I suppose however that this is not an issue too much, as likely at some point the user will come back to the first document to finish their work and hence diagnostics will be calculated for the initial document.